Publishing the master build is failing when installing mkdocs dependencies

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
After fixing #10281, the [publish build action](https://github.com/rook/rook/runs/6497822436?check_suite_focus=true) is failing when building the docs due to the error:
```
No matching distribution found for mkdocs-awesome-pages-plugin (from -r requirements_docs.txt (line 3))
```

A longer snippet of the log:
```
Must remake target 'build.images'.
  Successfully remade target file 'build.images'.
  Considering target file 'build.docs'.
   File 'build.docs' does not exist.
   Finished prerequisites of target file 'build.docs'.
  Must remake target 'build.docs'.
  Could not find a version that satisfies the requirement mkdocs-awesome-pages-plugin (from -r requirements_docs.txt (line 3)) (from versions: )
make: Entering directory '/home/runner/work/rook/rook/build/release'
No matching distribution found for mkdocs-awesome-pages-plugin (from -r requirements_docs.txt (line 3))
pip install -r requirements_docs.txt
make: *** [build.docs] Error 1
Collecting mike (from -r requirements_docs.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/2d/76/2d89ba71dcf8e7a74b5e152fc8e86fe7c1f37bd86563fa8dea927f521425/mike-1.1.2.tar.gz
Collecting mkdocs (from -r requirements_docs.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/db/f9/b0179afee0db[2194](https://github.com/rook/rook/runs/6497822436?check_suite_focus=true#step:8:2194)3120ea606eb68bda1257b96420df74b775280eb5850b/mkdocs-1.0.4-py2.py3-none-any.whl (1.2MB)
Collecting mkdocs-awesome-pages-plugin (from -r requirements_docs.txt (line 3))
Makefile:135: recipe for target 'build.docs' failed
make: Leaving directory '/home/runner/work/rook/rook/build/release'
Error: Process completed with exit code 2.
```

**Expected behavior:**
Publishing the build should succeed.

Reopening for two follow-up issues:
1. Since updating the publish workflow to ubuntu-20.04, the running is failing because it's an arm64 arch. 

```
Successfully remade target file 'list-image'.
  Successfully remade target file 'build.common'.
Makefile:137: *** cross platform image build only supported on amd64 host currently.  Stop.
 Finished prerequisites of target file 'build.all'.
Must remake target 'build.all'.
Error: Process completed with exit code 2.
```
Connecting to the runner:
```
ubuntu@ubuntu-focal-aarch64-01:~/actions-runner/_work/rook/rook$ printenv | grep ARCH
RUNNER_ARCH=ARM64
```

2. The release-1.8 build publish is failing, we must need to backport the new docs publishing there as well. For example see the failure [here](https://github.com/rook/rook/actions/runs/2354484680) after a backport merged.

```
Cloning into '/home/runner/work/rook/rook/.work/rook.github.io'...
rm -rf /home/runner/work/rook/rook/.work/rook.github.io
cp: cannot create directory '/home/runner/work/rook/rook/.work/rook.github.io/docs/rook/v1.8': No such file or directory
mkdir -p /home/runner/work/rook/rook/.work/rook.github.io
git clone --depth=1 -b master https://***@github.com/rook/rook.github.io.git /home/runner/work/rook/rook/.work/rook.github.io
rm -rf /home/runner/work/rook/rook/.work/rook.github.io/docs/rook/v1.8
cp -r /home/runner/work/rook/rook/Documentation/ /home/runner/work/rook/rook/.work/rook.github.io/docs/rook/v1.8
Makefile:127: recipe for target 'build.docs' failed
make: Leaving directory '/home/runner/work/rook/rook/build/release'
make: *** [build.docs] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/push-build.yaml
+++ b/.github/workflows/push-build.yaml
@@ -14,7 +14,7 @@ defaults:
 
 jobs:
   push-image-to-container-registry:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-18.04
     if: github.repository == 'rook/rook'
     steps:
       - name: checkout
@@ -50,11 +50,6 @@ jobs:
         run: |
           echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
           echo "GITHUB_REF"=${GITHUB_REF} >> $GITHUB_ENV
-
-      - name: pip install
-        run: |
-          sudo apt-get update
-          sudo apt-get -y install python3-pip
 
       - name: build and release
         env:

